vendor/github.com/go-openapi/strfmt/time.go

```diff
@@ -56,12 +55,14 @@ const (
 	RFC3339Millis = "2006-01-02T15:04:05.000Z07:00"
 	// RFC3339Micro represents a ISO8601 format to micro instead of to nano
 	RFC3339Micro = "2006-01-02T15:04:05.000000Z07:00"
+	// ISO8601LocalTime represents a ISO8601 format to ISO8601 in local time (no timezone)
+	ISO8601LocalTime = "2006-01-02T15:04:05"
 	// DateTimePattern pattern to match for the date-time format from http://tools.ietf.org/html/rfc3339#section-5.6
 	DateTimePattern = `^([0-9]{2}):([0-9]{2}):([0-9]{2})(.[0-9]+)?(z|([+-][0-9]{2}:[0-9]{2}))$`
 )
 
 var (
-	dateTimeFormats = []string{RFC3339Micro, RFC3339Millis, time.RFC3339, time.RFC3339Nano}
+	dateTimeFormats = []string{RFC3339Micro, RFC3339Millis, time.RFC3339, time.RFC3339Nano, ISO8601LocalTime}
```

Comment:
this adds to the accepted formats the `datetime` validator permits, the docs added in https://github.com/kubernetes/kubernetes/pull/85381/files#r347187064 are not complete

Reply:
we can easily override the time format, copy their regexps and that way stop adding ISO8601LocalTime.

Reply:
The validation code path does not use the `ISO8601LocalTime` regex, but only the one for rfc3339. This means that the `datetime` format in our API rejects e.g. timezone-less `2006-01-02T15:04:05`, but also the mille and micro second variant. I don't know why the upstream project added that support to their unmarshaller.

Reply:
With that, I think this PR is fine.

Reply:
can you open a followup issue to track the override/cleanup work we discussed:
* pinning the named/supported validators we currently have
* forking/extracting the openapi validator